File: app/src/main/java/com/ytempest/wanandroid/activity/main/project/ProjectClassifyAdapter.java
package com.ytempest.wanandroid.activity.main.project;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import com.ytempest.wanandroid.activity.main.project.content.ProjectContentFrag;
import com.ytempest.wanandroid.http.bean.ProjectClassifyBean;
import com.ytempest.wanandroid.utils.CoreFragPagerAdapter;

/**
 * @author heqidu
 * @since 2020/12/25
 */
public class ProjectClassifyAdapter extends CoreFragPagerAdapter<ProjectClassifyBean> {

    public ProjectClassifyAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    protected Fragment onCreateFragment(ProjectClassifyBean data, int pos) {
        return ProjectContentFrag.newInstance(data);
    }

}

File: app/src/main/java/com/ytempest/wanandroid/activity/main/project/ProjectClassifyAdapter.kt
package com.ytempest.wanandroid.activity.main.project

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import com.ytempest.wanandroid.activity.main.project.content.ProjectContentFrag

import com.ytempest.wanandroid.http.bean.ProjectClassifyBean
import com.ytempest.wanandroid.utils.CoreFragPagerAdapter

/**
 * @author heqidu
 * @since 21-2-10
 */
class ProjectClassifyAdapter(
        fm: FragmentManager
) : CoreFragPagerAdapter<ProjectClassifyBean>(fm) {

    override fun onCreateFragment(data: ProjectClassifyBean, pos: Int): Fragment {
        return ProjectContentFrag.newInstance(data);
    }
}